zone/npc_scale_manager.cpp

```diff
@@ -487,14 +536,21 @@ bool NpcScaleManager::IsAutoScaled(NPC* npc)
  */
 bool NpcScaleManager::ApplyGlobalBaseScalingToNPCStatically(NPC *&npc)
 {
-	int8 npc_type  = GetNPCScalingType(npc);
-	int  npc_level = npc->GetLevel();
-
-	global_npc_scale g = GetGlobalScaleDataForTypeLevel(npc_type, npc_level);
+	auto npc_type         = GetNPCScalingType(npc);
+	auto npc_level        = npc->GetLevel();
+	auto zone_id          = npc->GetZoneID();
```

Comment:
This won't find anything from the base tables and breaks original functionality if we assume `zone_id` 0

Code will need to look for the specific zone first and then try zone_id 0 after if it can't find it